src/lib.rs

```diff
@@ -2169,9 +2187,17 @@ unsafe impl<T> Sync for IterMut<'_, T> where T: Sync {}
 /// Creates the initial generation seeded by the current time.
 #[must_use]
 fn create_initial_generation() -> u64 {
-  let mut hasher = RandomState::new().build_hasher();
-  hasher.write_u32(0);
-  hasher.finish()
+  #[cfg(feature = "std")]
+  {
+    use std::{collections::hash_map::RandomState, hash::BuildHasher};
+
+    let mut hasher = RandomState::new().build_hasher();
+    hasher.write_u32(0);
+    hasher.finish()
+  }
+
+  #[cfg(not(feature = "std"))]
+  const_random::const_random!(u64)
```

Comment:
I fixed it to use a static Xorshift algorithm in combination with the `const_random`. However, if the idea is to have a different number for each list, then the current algorithm is broken, since it can be called twice at the exact same instant from two different threads. You can fix it by also hashing the current thread's ID.

Reply:
Yeah guaranteeing a random number was never the goal, it's more of a best effort thing to reduce the possibility of bugs occurring due to misuse of `Index`. The xorshift approach should be sufficient for no_std.